wallet.js

```diff
@@ -0,0 +1,229 @@
+    this.password = '123';
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    console.log('Websocket connection:', value);
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      hathorLib.wallet.loadAddressHistory(0, hathorLib.constants.GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }); // TODO Catch exception.
+    }
+  }
+
+  getCurrentAddress() {
+    return hathorLib.wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      // TODO we also have to update some wallet lib data? Lib should do it by itself
+      const walletData = hathorLib.wallet.getWalletData();
+      const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+      const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+      hathorLib.wallet.updateHistoryData(historyTransactions, allTokens, [wsData.history], null, walletData);
+
+      const newWalletData = hathorLib.wallet.getWalletData();
+      const { keys } = newWalletData;
+      //this.props.newTx(wsData.history, keys);
+
+      this.emit('new-tx', wsData.history);
+    }
+  }
+
+  reloadData() {
+    console.log('reloadData');
+  }
+
+  getAllBalances() {
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(tx, addresses) {
+    const updatedHistoryMap = {};
+    const updatedBalanceMap = {};
+    const balances = this.getTxBalance(tx);
+
+    // we now loop through all tokens present in the new tx to get the new history and balance
+    for (const [tokenUid, tokenTxBalance] of Object.entries(balances)) {
+      // we may not have this token yet, so state.tokensHistory[tokenUid] would return undefined
+      const currentHistory = state.tokensHistory[tokenUid] || [];
+      const newTokenHistory = addTxToSortedList(tokenUid, tx, tokenTxBalance, currentHistory);
+      updatedHistoryMap[tokenUid] = newTokenHistory;
+      // totalBalance should not be confused with tokenTxBalance. The latter is the balance of the new
+      // tx, while the former is the total balance of the token, considering all tx history
+      const totalBalance = getBalance(tokenUid);
+      updatedBalanceMap[tokenUid] = totalBalance;
+    }
+    const newTokensHistory = Object.assign({}, state.tokensHistory, updatedHistoryMap);
+    const newTokensBalance = Object.assign({}, state.tokensBalance, updatedBalanceMap);
+  };
+
+  /**
+   * Send tokens to only one address.
+   **/
+  sendTransaction(address, value, token) {
+    const isHathorToken = token.uid === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+    const data = {
+      tokens: isHathorToken ? [] : [token.uid],
+      inputs: [],
+      outputs: [{
+        address, value, tokenData: 0,
```

Comment:
`tokenData` should be `isHathorToken ? 0 : 1`

Reply:
I just used `token.uid`. What do you think?

Reply:
The `tokenData` is the index of this token in the `tokens` list. If it's a HTR token, we use 0, otherwise we use the `index - 1`

Reply:
Hum, you're right. I'll fix it. Thanks!

Reply:
Fixed. Thanks!